### PR TITLE
Setting Broker TargetFramework to netstandard2.0

### PR DESCRIPTION
**Pull request by gladjohn**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-
     <TargetFrameworkNetDesktop>net462</TargetFrameworkNetDesktop>
     <TargetFrameworkNetCore>netcoreapp3.1</TargetFrameworkNetCore>
+    <TargetFrameworkNetStandard>netstandard2.0</TargetFrameworkNetStandard>
 
     <PlatformTarget>AnyCPU</PlatformTarget>
-    <TargetFrameworks Condition="$([MSBuild]::IsOsPlatform('Windows'))">$(TargetFrameworkNetDesktop);$(TargetFrameworkNetCore)</TargetFrameworks>
-    <TargetFramework Condition="$([MSBuild]::IsOsPlatform('OSX'))">$(TargetFrameworkNetCore)</TargetFramework>
-    <TargetFramework Condition="$([MSBuild]::IsOsPlatform('Linux')) ">$(TargetFrameworkNetCore)</TargetFramework>
+    <TargetFramework Condition="$([MSBuild]::IsOsPlatform('Windows'))">$(TargetFrameworkNetStandard)</TargetFramework>
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client.Broker/Microsoft.Identity.Client.Broker.csproj:8]: nit: There's no difference between lines 8, 9 and 10. You can simply have ` <TargetFramework>$(TargetFrameworkNetStandard)</TargetFramework>`

File: tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj [hunk 2]
```diff
@@ -49,7 +49,7 @@
     <PackageReference Include="Microsoft.Windows.SDK.Contracts" Version="10.0.19041.1" />
     <ProjectReference Include="..\..\src\client\Microsoft.Identity.Client.Desktop\Microsoft.Identity.Client.Desktop.csproj" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1' ">
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard1.3' ">
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj:52]: Why does this one need to change?
The unit tests only use net core and net framework?

Comment (gladjohn) [reply tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj:52]: This is just for `Microsoft.Identity.Client.Broker` as it does not have a target for `netcoreapp3.1`

Comment (pmaytak) [reply tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj:52]: Does it have to be netstandard2.0 then or is this based on MSAL's netstandard1.3?

File: src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs [hunk 3]
```diff
@@ -295,7 +295,7 @@ public AcquireTokenInteractiveParameterBuilder WithParentActivityOrWindow(IWin32
         }
 #endif
 
-#if DESKTOP || NET5_WIN || NET_CORE
+#if DESKTOP || NET5_WIN || NET_CORE || NETSTANDARD
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs:298]: Why are these changes needed?

File: tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj (hunk 2 above)
Comment (gladjohn) [reply tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj:52]: this is based on `<TargetFrameworkNetStandard>netstandard1.3</TargetFrameworkNetStandard>` in `Microsoft.Identity.Client`

File: src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs (hunk 3 above)
Comment (gladjohn) [reply src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs:298]: Broker target is NetStandard and these are consumed by broker.

Comment (pmaytak) [reply src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs:298]: Where at? I didn't find any usages. Guess just trying to figure out the order in which targets are chosen. I thought main Msal target would be determined by my app's target not Msal.Broker.

Also for some reason Msal.Desktop doesn't build for me (VS 19 and 22) when building the Net Core WAM dev app...
image

File: tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj (hunk 2 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Unit/Microsoft.Identity.Test.Unit.csproj:52]: This doesn't look right. Our tests run against: 

    <TargetFrameworkNetDesktop461>net472</TargetFrameworkNetDesktop461>
    <TargetFrameworkNetCore>netcoreapp3.1</TargetFrameworkNetCore>
    <TargetFrameworkNet5Win>net5.0-windows10.0.17763.0</TargetFrameworkNet5Win>

None of them run against `'netstandard1.3'` so this item group would simply be ignored.

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
Unit test csproj changes

File: src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs (hunk 3 above)
Comment (gladjohn) [reply src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs:298]: taking a look at it, I am not able to repro it still. DevApps seems to building for me.

**Review by bgavrilMS — APPROVED** [review]

**Review by SameerK-MSFT — APPROVED** [review]
We want to make sure that samples are compatible with the changes.
